### ISSUE TITLE
Support for xml namespace

### ISSUE THREAD

**Issue by sockeqwe** [Tickaroo/tikxml#9]
Not really needed by us. However, some people might find it useful.

Ok, so after having thought about namespace support, this is what I cam up with:

We are going to have limited namespace support by allowing the user to write `:` in the names.
This means we only support STABLE namespaces (not dynamically changeable) as we simply treat namespaces + element names / attribute names as one single name (string). Furthermore, namespace can't be implemented dynamically (read dynamically namespace and prefix while parsing xml) as annotation processor has already to know which
- [x] support for`@Path`. Already implemented with #38 
- [x] unit tests for `@Attribute(name = "myNamespace:foo")`, `@PropertyElement(name = "myNamespace:foo")`, `@Element(name = "myNamespace:foo")`
- [x] ignore namespace definitions like: `xmlns:m="http://www.w3.org/1998/Math/MathML"` so that `.exceptionOnUnreadXml(true)` will not throw an exception when reading namespace definitions.
- [x] Add a `@NamespaceDefinition(prefix="m", namespaceName="http://www.w3.org/1998/Math/MathML")` annotation that will be used when writing xml to add a namespace definition. Finde a better name for the annotation as the annotation will only be used when writing xml (has no impact to reading/parsing xml). The annotation name should state this out (only take into account while writing xml)

In a second step (maybe TikXml 2.0), we could also try to add full dynamic namespace support. i.e. we could read namespace prefixes while reading / parsing xml and update internal mapping of `TypeAdapter's`internal HashMaps for xml attribute names and xml elements  as we read the namespaces ... but that is out of scope in version 1.0 as I'm pretty sure "stable" namespace support is enough for most users. Also, namespaces can be overridden in any child node at any time, which will effect all involved TypeAdapters. Maybe some kind of observer pattern will be needed to inform other TypeAdpters about overriden namespaces. To sum it up: not an easy task (doable though) and I'm not sure if it is worth the effort.

---

The original plan was to use annotations to represent namespaces, **but it turns out that this is not possible with `@Path()` annotations like `@Path("x:foo/y:bar")` where we have 2 namespaces.**

So this idea has been dropped!

The original idea was: 
Would be nice to have namespaces annotation based, something like 

``` java
@Namespace("foo")
public @interface FooNamespace{}

@Xml
class Something {
     @FooNamespace
     @PropertyElement String name;
}
```

**Comment by sockeqwe** [comment]
Turns out annotation based aproach wont work well with `@Path`

**Comment by sockeqwe** [comment]
Ok, so after having thought about namespace support, this is what I cam up with:

We are going to have limited namespace support by allowing the user to write `:` in the names.
This means we only support STABLE namespaces (not dynamically changeable) as we simply treat namespaces + element names / attribute names as one single name (string). Furthermore, namespace can't be implemented dynamically (read dynamically namespace and prefix while parsing xml) as annotation processor has already to know which

- [x] support for`@Path`. Already implemented with #38 
- [ ] unit tests for `@Attribute(name = "myNamespace:foo")`, `@PropertyElement(name = "myNamespace:foo")`, `@Element(name = "myNamespace:foo")`
- [ ] ignore namespace definitions like: `xmlns:m="http://www.w3.org/1998/Math/MathML"` so that `.exceptionOnUnreadXml(true)` will not throw an exception when reading namespace definitions.
- [ ] Add a `@NamespaceDefinition(prefix="m", namespaceName="http://www.w3.org/1998/Math/MathML")` annotation that will be used when writing xml to add a namespace definition. Finde a better name for the annotation as the annotation will only be used when writing xml (has no impact to reading/parsing xml). The annotation name should state this out (only take into account while writing xml)
